woodwork/table_accessor.py

```diff
@@ -198,8 +198,7 @@ def __setitem__(self, col_name, column):
             warnings.warn("Cannot allow adding column with index tags. Tag removed", IndexTagRemovedWarning)
             column.ww.remove_semantic_tags('index')
             logical_type = column.ww.logical_type
-            if hasattr(logical_type, 'standard_tags'):
-                column.ww.add_semantic_tags(logical_type.standard_tags)
+            column.ww.add_semantic_tags(logical_type.standard_tags)
```

Comment:
one more behavior thing - we allow users to set `use_standard_tags` for a column to `False` which will stop a column that would normally have the `numeric` tag, for example, from having it. I believe that with this implementation, `column.ww.use_standard_tags` could be False but it'd still get the standard tags.

One option would be to add onto the current implementation, and only perform this line if `column.ww.use_standard_tags is True`, but I also wanted to present another idea that would allow us to combine this line and the two above into one line:

`column.ww.set_semantic_tags(column.ww.semantic_tags - {'index'})` 

`set_semantic_tags` also takes care of the standard tag logic (https://github.com/alteryx/woodwork/blob/main/woodwork/column_schema.py#L179) which is why this lets us simplify here. 

Whichever implementation you go with, we should add an additional test function that checks that we don't have the standard tags if `use_standard_tags` is False. (use_standard_tags can be set at woodwork init for dataframe or series)